analyzers/src/SonarAnalyzer.CSharp/Rules/ExpectedExceptionAttributeShouldNotBeUsed.cs

```diff
@@ -27,6 +27,10 @@ public sealed class ExpectedExceptionAttributeShouldNotBeUsed : ExpectedExceptio
     {
         protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
 
+        protected override bool HasExpectedExceptionAttribute(SyntaxNode node) =>
+            node is MethodDeclarationSyntax methodDeclaration
```

Comment:
Missing coverage on one condition here.
I don't think I can improve it 😞

Reply:
You can use a cast instead of the is operator: `((MethodDeclarationSyntax)node).AttributeLists.Any(...);`
You only register for method declarations, so the cast will never fail.